x-pack/legacy/plugins/file_upload/public/components/json_index_file_picker.js

```diff
@@ -21,89 +21,90 @@ export class JsonIndexFilePicker extends Component {
   state = {
     fileUploadError: '',
     fileParsingProgress: '',
-    fileRef: null
   };
 
-  componentDidUpdate(prevProps, prevState) {
-    if (prevState.fileRef !== this.props.fileRef) {
-      this.setState({ fileRef: this.props.fileRef });
-    }
+  async componentDidMount() {
+    this._isMounted = true;
   }
 
-  _fileHandler = async fileList => {
-    const {
-      resetFileAndIndexSettings, setParsedFile, onFileRemove, onFileUpload,
-      transformDetails, setFileRef, setIndexName
-    } = this.props;
-
-    const { fileRef } = this.state;
+  componentWillUnmount() {
+    this._isMounted = false;
+  }
 
-    resetFileAndIndexSettings();
+  _fileHandler = fileList => {
+    const fileArr = Array.from(fileList);
+    this.props.resetFileAndIndexSettings();
     this.setState({ fileUploadError: '' });
-    if (fileList.length === 0) { // Remove
-      setParsedFile(null);
-      if (onFileRemove) {
-        onFileRemove(fileRef);
-      }
-    } else if (fileList.length === 1) { // Parse & index file
-      const file = fileList[0];
-      if (!file.name) {
+    if (fileArr.length === 0) { // Remove
+      return;
+    }
+    if (fileArr.length === 1) { // Parse & index file
+      const file = fileArr[0];
+      let initIndexName;
+      try {
+        initIndexName = this._getIndexName(file);
+      } catch (error) {
         this.setState({
-          fileUploadError: i18n.translate(
-            'xpack.fileUpload.jsonIndexFilePicker.noFileNameError',
-            { defaultMessage: 'No file name provided' })
+          fileUploadError: i18n.translate('xpack.fileUpload.jsonIndexFilePicker.errorGettingIndexName', {
+            defaultMessage: 'Error retrieving index name: {errorMessage}',
+            values: {
+              errorMessage: error.message
+            }
+          })
         });
+        this.props.resetFileAndIndexSettings();
```

Comment:
Do you need to call resetFileAndIndexSettings here? Its getting called on line 36 above and it looks like else has happened that would require it to be called again.

Reply:
Not anymore. Removed!